pkg/api/validation/name.go

```diff
@@ -27,17 +27,26 @@ var NameMayNotBe = []string{".", ".."}
 // NameMayNotContain specifies substrings that cannot be used in names specified as path segments (like the REST API or etcd store)
 var NameMayNotContain = []string{"/", "%"}
 
-// ValidatePathSegmentName validates the name can be used as a path segment
-func ValidatePathSegmentName(name string, prefix bool) (bool, string) {
-	// Only check for exact matches if this is the full name (not a prefix)
-	if prefix == false {
-		for _, illegalName := range NameMayNotBe {
-			if name == illegalName {
-				return false, fmt.Sprintf(`name may not be %q`, illegalName)
-			}
+// IsValidPathSegmentName validates the name can be safely encoded as a path segment
+func IsValidPathSegmentName(name string) (bool, string) {
+	for _, illegalName := range NameMayNotBe {
+		if name == illegalName {
+			return false, fmt.Sprintf(`name may not be %q`, illegalName)
+		}
+	}
+
+	for _, illegalContent := range NameMayNotContain {
+		if strings.Contains(name, illegalContent) {
+			return false, fmt.Sprintf(`name may not contain %q`, illegalContent)
 		}
 	}
 
+	return true, ""
+}
+
+// IsValidPathSegmentPrefix validates the name can be used as a prefix for a name which will be encoded as a path segment
+// It does not check for exact matches with disallowed names, since an arbitrary suffix might make the name valid
+func IsValidPathSegmentPrefix(name string) (bool, string) {
```

Comment:
to comply with the name validation required by ValidateObjectMeta, implemented here: https://github.com/kubernetes/kubernetes/pull/16717/files#diff-b72277cd2bba86ca73e54d9589bdf5f9R62